packages/blade/src/components/Dropdown/_decisions/decisions.md

```diff
@@ -0,0 +1,697 @@
+        <Button>Primary Button</Button>
+      </>
+    }
+  />
+</ActionList>
+```
+
+<details>
+<summary>Alternate Prop-based API</summary>
+
+**Props and constrained**
+
+```jsx
+<ActionList surfaceLevel={2}>
+  <ActionListHeader title="Recent Searches" icon={HistoryIcon} />
+  <ActionListSectionHeading title="Hello" />
+  <ActionListItem
+    title="Item Value"
+    desciption="Item Description"
+    leading={StarIcon} // Can be IconComponent type or Image URL
+    trailing="⌘ + Space"
+  />
+  <ActionListDivider />
+  <ActionListFooter
+    title="Footer Title"
+    description="Footer Description"
+    leading={DocIcon}
+    trailing={{
+      primary: {
+        onClick: () => {},
+        text: 'Primary Action',
+      },
+    }}
+    // OR
+    trailing={ArrowRightIcon}
+    // OR
+    trailing={{
+      primary: {
+        onClick: () => {
+          console.log('Feedback Yes Clicked');
+        },
+        icon: CheckIcon,
+      },
+      secondary: {
+        onClick: () => {
+          console.log('Feedback No Clicked');
+        },
+        icon: CrossIcon,
+      },
+    }}
+  />
+</ActionList>
+```
+
+</details>
+
+### `ActionList` Parent Component
+
+Wrapper for all ActionList compound components. Should handle the logic of selections.
+
+#### API
+
+Usage
+
+```jsx
+<ActionList>{/* Child components */}</ActionList>
+```
+
+Props
+
+```ts
+type ActionListProps = {
+  children: React.ReactNode;
+
+  /**
+   * Renders simple menu or checkboxes depending on the variant
+   *
+   * @default single
+   */
+  variant?: 'single' | 'multiple';
+
+  /** Callback that gets called when item is checked/clicked */
+  onChange?: () => void;
+
+  /**
+   * Changes the background surface level color
+   *
+   * @default 2
+   */
+  surfaceLevel: 2 | 3;
+};
+```
+
+### `ActionListItem`
+
+![](2022-11-29-17-29-01.png)
+
+#### Design
+
+- [Figma - ActionListItem / Body](https://www.figma.com/file/sAdplk2uYnI2ILnDKUxycW/Blade---Banking-Dark?node-id=13136%3A378703&t=RoEboUS3qWdStKK6-0)
+
+#### API
+
+Usage
+
+```jsx
+<ActionListItem
+  title="Accept API"
+  value="accept"
+  desciption="Accept this API"
+  leading={<ActionListItemIcon icon={StarIcon} />}
+  intent="negative"
+  trailing={<ActionListItemText>⌘ + Space + A</ActionListItemText>}
+  isDefaultSelected={false}
+  onClick={() => {
+    console.log('Item clicked');
+  }}
+/>
+```
+
+Props
+
+```ts
+type ActionListItemProps = {
+  title: string;
+
+  description: string;
+
+  /** Value that we use in JS form. Similar to `value` in `Checkbox` */
+  value: string;
+
+  /**
+   * Leading Icon or assets. Accepts JSX Component
+   *
+   * Should not be allowed when ActionList variant is multiple select
+   */
+  leading?: React.ReactNode;
+
+  /**
+   * Trailing Assets. Should ideally be ActionListItemText component for now
+   */
+  trailing?: React.ReactNode;
+
+  /**
+   * Changes item color to red
+   *
+   * @default neutral
+   */
+  intent?: 'negative' | 'neutral';
```

Comment:
is our neutral color same as feedback neutral? for eg: negative is red in tokens but is neutral color here same as feedback.neutra.text?

Reply:
ohhh nope its surface.text.normal. Lemme discuss this with design and see if we should some use some other prop instead of intent here then 🤔

Reply:
Actually it seems valid to have surface.text.normal in normal so I'll change it here.

Reply:
TLDR: so I had added `<ActionListItem intent="negative" />` which uses feedback negative token and we have the default variant which uses `surface.text.normal`

we don't exactly have intent="normal" anywhere else so I changed it to variant="negative | normal"

we had [intent vs variant discussion](https://github.com/razorpay/blade/blob/89cc8babfe8d37a29354f7f3bc7689a556b1bcc7/packages/blade/src/components/Badge/_decisions/decisions.md#variant-prop) during badge API decisions so referring to that for this where we decided to use intent for semantic values and variant for everything else.

Reply:
what if we just have `intent?: negative` so if it's undefined we set it to normal by default and if someone passes intent then the only value we support right now is negative.

Reply:
hmm yes that would also make sense. Lets go with that only then.